### PR TITLE
add a note that describes grain rebuilding on restart and refresh

### PR DESCRIPTION
**Pull request by whytewolf**
### What does this PR do?

Documentation enhancement to describe grain refreshing, that is commonly understood to expert users but can be missed by newer users when writing custom grains. 

### What issues does this PR fix or reference?

Understanding of grain rebuilding. 

### Tests written?

No - Documentation

### Commits signed with GPG?

No 

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by cachedout — CHANGES_REQUESTED** [review]
This is not true. See `grains_cache` in the salt codebase.

**Comment by whytewolf** [comment]
okay, I can add an 'unless `grains_cache` is set to true' I think that would clear it up. because my statement isn't completely untrue in a default config. at least until `grains_cache` defaults to true.

**Comment by whytewolf** [comment]
Is this change better? describing how it operates by default and the option you pointed out.

**Comment by whytewolf** [comment]
zd-1837

**Review by cachedout — CHANGES_REQUESTED** [review]

File: doc/topics/grains/index.rst [hunk 1]
```diff
@@ -298,3 +298,9 @@ Syncing grains can be done a number of ways, they are automatically synced when
 above) the grains can be manually synced and reloaded by calling the
 :mod:`saltutil.sync_grains <salt.modules.saltutil.sync_grains>` or
 :mod:`saltutil.sync_all <salt.modules.saltutil.sync_all>` functions.
+
+.. note::
+
+    When the :conf_minion:`grains_cache` is set to False, the grains dictionary is built
+    and stored in memory on the minion. Every time the minion restarts or
+    `saltutil.refresh_grains` is run, the grain dictionary is rebuilt from scratch.
```
Comment (rallytime) [inline doc/topics/grains/index.rst]: You'll need double back-ticks on references like `saltutil.refresh_grains` to render correctly in restructured text syntax. (Mark-down is single back-ticks for code references, RST is double back-ticks.)

Comment (whytewolf) [reply doc/topics/grains/index.rst]: okay, going to add that now. so used to MD lately forgot this was in RST. Thank you. change committed.

**Review by rallytime — APPROVED** [review]

**Pull request merged**